scripts/prepare_transport_data_input.py

```diff
@@ -71,40 +102,38 @@ def download_CO2_emissions():
     The dataset is downloaded from the following link: https://data.worldbank.org/indicator/EN.CO2.TRAN.ZS?view=map
     It is until the year 2014. # TODO: Maybe search for more recent years.
     """
-    url = (
-        "https://api.worldbank.org/v2/en/indicator/EN.CO2.TRAN.ZS?downloadformat=excel"
-    )
 
-    # Read the 'Data' sheet directly from the Excel file at the provided URL
     try:
+        url = "https://api.worldbank.org/v2/en/indicator/EN.CO2.TRAN.ZS?downloadformat=excel"
+
         CO2_emissions = pd.read_excel(url, sheet_name="Data", skiprows=[0, 1, 2])
-        print("File read successfully.")
-    except Exception as e:
-        print("Failed to read the file:", e)
-        return pd.DataFrame()
 
-    CO2_emissions = CO2_emissions[
-        ["Country Name", "Country Code", "Indicator Name", "2014"]
-    ]
+        CO2_emissions = CO2_emissions[
+            ["Country Name", "Country Code", "Indicator Name", "2014"]
+        ]
 
-    # Calculate efficiency based on CO2 emissions from transport (% of total fuel combustion)
-    CO2_emissions["average fuel efficiency"] = (100 - CO2_emissions["2014"]) / 100
+        # Calculate efficiency based on CO2 emissions from transport (% of total fuel combustion)
+        CO2_emissions["average fuel efficiency"] = (100 - CO2_emissions["2014"]) / 100
 
-    # Add ISO2 country code for each country
-    CO2_emissions = CO2_emissions.rename(columns={"Country Name": "Country"})
-    cc = coco.CountryConverter()
-    CO2_emissions.loc[:, "country"] = cc.pandas_convert(
-        series=CO2_emissions["Country"], to="ISO2", not_found="not found"
-    )
+        CO2_emissions = CO2_emissions.dropna(subset=["average fuel efficiency"])
 
-    # Drop region names that have no ISO2:
-    CO2_emissions = CO2_emissions[CO2_emissions.country != "not found"]
+        CO2_emissions = CO2_emissions.rename(columns={"Country Name": "Country"})
 
-    # Drop region names where country column contains list of countries
-    CO2_emissions = CO2_emissions[
-        CO2_emissions.country.apply(lambda x: isinstance(x, str))
-    ]
-    return CO2_emissions
+        CO2_emissions = _add_iso2_code_per_country_and_clean_data(CO2_emissions)
+
+        CO2_emissions["average fuel efficiency"] = CO2_emissions[
+            "average fuel efficiency"
+        ].astype(float)
+
+        CO2_emissions.loc[:, "average fuel efficiency"] = CO2_emissions[
+            "average fuel efficiency"
+        ].round(3)
+
+        return CO2_emissions[["country", "average fuel efficiency"]]
+
+    except Exception as e:
+        logger.warning("Failed to read the file:", e)
+        return pd.DataFrame()
```

Comment:
Moving the exception at the end may mask some errors that happen along the procedure due to change in the data format, rather than access to the source.
That may be less desirable, is there a reason for the proposal?